src/cocotb_tools/runner.py

```diff
@@ -1609,6 +1609,103 @@ def _test_command(self) -> List[_Command]:
         return cmds
 
 
+class Dsim(Runner):
+    """Implementation of :class:`Runner` for DSim.
+
+    * Does not support the ``gui`` argument to :meth:`.test`.
+    * Does not support the ``pre_cmd`` argument to :meth:`.test`.
+    """
+
+    supported_gpi_interfaces = {"verilog": ["vpi"]}
+
+    def _simulator_in_path(self) -> None:
+        if shutil.which("dsim") is None:
+            raise SystemExit("ERROR: dsim executable not found!")
+
+    def _get_include_options(self, includes: Sequence[PathLike]) -> _Command:
+        return [f"+incdir+{_as_tcl_value(str(include))}" for include in includes]
+
+    def _get_define_options(self, defines: Mapping[str, object]) -> _Command:
+        return [
+            f"+define+{name}={_as_sv_literal(value)}" for name, value in defines.items()
+        ]
+
+    def _get_parameter_options(self, parameters: Mapping[str, object]) -> _Command:
+        return [
+            f"-defparam {name}={_as_sv_literal(value)}"
+            for name, value in parameters.items()
+        ]
+
+    @property
+    def sim_file(self) -> Path:
+        return self.build_dir / "image.so"
+
+    def _test_command(self) -> List[_Command]:
+        plusargs = self.plusargs
+        if self.waves:
+            plusargs += ["-waves file.vcd"]
+
+        if self.pre_cmd is not None:
+            raise ValueError("WARNING: pre_cmd is not implemented for DSim.")
```

Comment:
This should probably be `NotImplementedError`. If the other sims use `ValueError` you can just ignore this suggestion and we'll clean them all up at the same time later.

Reply:
Hm, some simulators have `RuntimeError` and others have `ValueError`.

Reply:
Oh well.